src/Raven.Server/Documents/Revisions/DeleteRevisionsByChangeVectorMergedCommand.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.ServerWide.Context;
+using Voron.Data.Tables;
+using Voron;
+
+namespace Raven.Server.Documents.Revisions;
+public partial class RevisionsStorage
+{
+    internal sealed class DeleteRevisionsByChangeVectorMergedCommand : MergedTransactionCommand<DocumentsOperationContext, DocumentsTransaction>
+    {
+        private readonly string _id;
+
+        private readonly List<string> _cvs;
+
+        private readonly bool _includeForceCreated;
+
+        public long? Result;
+
+        public DeleteRevisionsByChangeVectorMergedCommand(string id, List<string> cvs, bool includeForceCreated)
+        {
+            _id = id;
+            _cvs = cvs;
+            _includeForceCreated = includeForceCreated;
+        }
+
+        protected override long ExecuteCmd(DocumentsOperationContext context)
+        {
+            Result = DeleteRevisions(context);
+            return 1;
+        }
+
+        private long? DeleteRevisions(DocumentsOperationContext context)
+        {
+            var revisionsStorage = context.DocumentDatabase.DocumentsStorage.RevisionsStorage;
+            var deleted = 0L;
+
+            using (DocumentIdWorker.GetSliceFromId(context, _id, out Slice lowerId))
+            using (revisionsStorage.GetKeyPrefix(context, lowerId, out Slice prefixSlice))
+            {
+                var collectionName = revisionsStorage.GetCollectionFor(context, prefixSlice);
+                if (collectionName == null)
+                {
+                    if (revisionsStorage._logger.IsInfoEnabled)
+                        revisionsStorage._logger.Info($"Tried to delete revisions for '{_id}' but no revisions found.");
+                    return 0;
+                }
+
+                var maxDeletes = revisionsStorage.GetRevisionsConfiguration(collectionName.Name, deleteRevisionsWhenNoCofiguration: true)
+                    .MaximumRevisionsToDeleteUponDocumentUpdate;
+
+                var result = revisionsStorage.ForceDeleteAllRevisionsForInternal(context, lowerId, prefixSlice, collectionName, maxDeletes, ShouldSkipRevision);
+                deleted += result.Deleted;
+            }
+
+            return deleted;
+        }
+
+        private bool ShouldSkipRevision(Document revision)
+        {
+            return ShouldSkipForceCreated(_includeForceCreated == false, revision.Flags) || _cvs.Contains(revision.ChangeVector) == false;
```

Comment:
why `_cvs.Contains(revision.ChangeVector)` ? you get revisions by change vector, so when can this not be the case?

Reply:
Look where I'm passing this method (`ForceDeleteAllRevisionsForInternal`)